annotation/tests/test_tasks_services.py

```diff
@@ -580,8 +789,1416 @@ def test_filter_tasks_db_no_files_or_jobs(
+    )
+    filter_mock.all.return_value = []
+
+    with patch(
+        "annotation.tasks.services.get_user_names_by_user_ids", return_value={}
+    ), patch(
+        "annotation.tasks.services.get_file_names_by_file_ids", return_value={}
+    ), patch(
+        "annotation.tasks.services.io.BytesIO"
+    ), patch(
+        "annotation.tasks.services.io.TextIOWrapper"
+    ):
+        with pytest.raises(HTTPException) as exc_info:
+            services.create_export_csv(mock_db, schema, "tenant", "token")
+
+        assert exc_info.value.status_code == 406
+        assert exc_info.value.detail == "Export data not found."
+
+
+def test_evaluate_agreement_score(
+    mock_session: Mock, mock_task: Mock, mock_tenant_data: Mock
+):
+    with patch(
+        "annotation.tasks.services.get_file_path_and_bucket",
+        return_value=(
+            "mock_s3_file_path",
+            "mock_s3_file_bucket",
+        ),
+    ) as mock_get_file_path_and_bucket, patch(
+        "annotation.tasks.services.get_agreement_score"
+    ) as mock_get_agreement_score, patch(
+        "annotation.tasks.services.compare_agreement_scores",
+        return_value=MagicMock(AgreementScoreComparingResult),
+    ) as mock_compare_agreement_scores:
+        mock_agreement_score_response = [
+            MagicMock(AgreementScoreServiceResponse)
+        ]
+        mock_get_agreement_score.return_value = mock_agreement_score_response
+        mock_session.query().all.return_value = [mock_task]
+        services.evaluate_agreement_score(
+            db=mock_session,
+            task=mock_task,
+            tenant="mock_tenant",
+            token=mock_tenant_data,
+        )
+
+        mock_get_file_path_and_bucket.assert_called_once_with(
+            mock_task.file_id, "mock_tenant", mock_tenant_data.token
+        )
+        mock_compare_agreement_scores.assert_called_once_with(
+            mock_agreement_score_response, services.AGREEMENT_SCORE_MIN_MATCH
+        )
+        mock_get_agreement_score.assert_called_once_with(
+            agreement_scores_input=[], tenant="mock_tenant", token="mock_token"
+        )
+
+
+def test_get_unique_scores(response_scores):
+    unique_scores = set()
+    task_id = 1
+    services.get_unique_scores(task_id, response_scores, unique_scores)
+    expected_scores = {
+        services._MetricScoreTuple(task_from=1, task_to=2, score=0.9),
+        services._MetricScoreTuple(task_from=1, task_to=3, score=0.7),
+    }
+
+    assert unique_scores == expected_scores
+
+
+def test_compare_agreement_scores_all_above_min_match(
+    mock_agreement_score_response: Mock,
+    mock_parse_obj_as: Mock,
+    mock_get_unique_scores: Mock,
+    mock_task_metric: Mock,
+    mock_agreement_score_comparing_result: Mock,
+):
+    min_match = 0.8
+    mock_parse_obj_as.return_value = [MagicMock(spec=ResponseScore)]
+    mock_task_metric.side_effect = (
+        lambda task_from_id, task_to_id, metric_score: MagicMock(
```

Comment:
https://docs.python.org/3/library/unittest.mock-examples.html#side-effect-functions-and-iterables

Reply:
use real data model